ilastik/experimental/api/_base.py

```diff
@@ -59,13 +59,28 @@ def predict(self, data):
                 )
 
             self._reorder_op.Input.setValue(data)
-
-            data = self._predict_op.PMaps.value[...]
+            roi = xarray_roi_to_slicing(xarray_roi, self._predict_op.PMaps.meta.axistags.keys())
+            data = self._predict_op.PMaps[roi].wait()
             return xarray.DataArray(data, dims=tuple(self._predict_op.PMaps.meta.axistags.keys()))
 
     return _PipelineImpl()
 
 
+def xarray_roi_to_slicing(xarray_roi, vigra_axistags):
+    if any(k not in vigra_axistags for k in xarray_roi.keys()):
+        raise ValueError(f"xarray_roi contains keys ({xarray_roi.keys()} not in axistags ({vigra_axistags})")
+    if not xarray_roi:
+        return Ellipsis
+
+    out_slicing = []
+    for axis in vigra_axistags:
+        if axis in xarray_roi:
+            out_slicing.append(xarray_roi[axis])
+        else:
+            out_slicing.append(slice(None, None, None))
+    return tuple(out_slicing)
```

Comment:
```suggestion
    return tuple(xarray_roi.get(axis, slice(None)) for axis in vigra_axistags)
```

Reply:
cheers, I was actually hoping to get this from xarrray and didn't look into their docs yet... :) But I'll use your suggestion if there is nothing to be found.